src/rust/en.mangairo/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+#![no_std]
+use aidoku::{
+	error::Result,
+	prelude::*,
+	std::{
+		net::{HttpMethod, Request},
+		String, Vec,
+	},
+	Chapter, DeepLink, Filter, Listing, Manga, MangaPageResult, Page,
+};
+
+mod parser;
+use parser::{BASE_URL, USER_AGENT};
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let mut result: Vec<Manga> = Vec::new();
+
+	let mut url = String::new();
+	parser::get_filtered_url(filters, page, &mut url);
+	let html = Request::new(url.as_str(), HttpMethod::Get).html()?;
+
+	let total_results = parser::parse_manga_list(html, &mut result);
+
+	if let Some(total_results_value) = total_results {
+		if total_results_value > result.len() as i32 * page {
+			return Ok(MangaPageResult {
+				manga: result,
+				has_more: true,
+			});
+		}
+	}
```

Comment:
instead of doing all this, I think you could do something like
```rust
let has_more = total_results.map_or(false, |value| value > result.len() as i32 * page);
```

Reply:
Thanks! I knew there had to be a better way but wasn't sure how to do it in Rust